src/vsr/replica.zig

```diff
@@ -6824,6 +7427,311 @@ pub fn ReplicaType(
+                self.commit_prepare = null;
+                self.commit_stage = .idle;
+            }
+
+            var grid_reads = self.grid_reads.iterate();
+            while (grid_reads.next()) |grid_read| {
+                self.message_bus.unref(grid_read.message);
+                self.grid_reads.release(grid_read);
+            }
+
+            var grid_writes = self.grid_writes.iterate();
+            while (grid_writes.next()) |grid_write| {
+                self.grid_writes.release(grid_write);
+            }
+
+            self.sync_dispatch(.requesting_target);
+        }
+
+        fn sync_request_trailers_callback(self: *Self) void {
+            assert(!self.solo());
+            assert(self.sync_stage == .request_trailers);
+            assert(!self.superblock.updating(.checkpoint));
+            assert(self.commit_stage == .idle);
+            assert(self.commit_prepare == null);
+            assert(!self.grid_repair_message_timeout.ticking);
+            assert(self.grid.read_queue.empty());
+            assert(self.grid.read_pending_queue.empty());
+            assert(self.grid.read_faulty_queue.empty());
+            assert(self.grid.write_queue.empty());
+            assert(self.grid_writes.executing() == 0);
+            if (self.status == .normal) assert(!self.primary());
+
+            const stage = self.sync_stage.request_trailers;
+            assert(stage.done());
+
+            // TODO(Zig) Use named format specifiers to avoid mixups.
+            log.debug("{}: sync_request_trailers_callback: " ++
+                "checkpoint_op={} checkpoint_id={x:0>32} " ++
+                "manifest_checksum={x:0>32} " ++
+                "free_set_checksum={x:0>32} " ++
+                "client_sessions_checksum={x:0>32}", .{
+                self.replica,
+                stage.target.checkpoint_op,
+                stage.target.checkpoint_id,
+                stage.manifest.final.?.checksum,
+                stage.free_set.final.?.checksum,
+                stage.client_sessions.final.?.checksum,
+            });
+
+            self.sync_dispatch(.{ .updating_superblock = .{
+                .target = stage.target,
+                .previous_checkpoint_id = stage.previous_checkpoint_id.?,
+                .checkpoint_op_checksum = stage.checkpoint_op_checksum.?,
+            } });
+        }
+
+        fn sync_superblock_update(self: *Self) void {
+            assert(!self.solo());
+            assert(self.sync_stage == .updating_superblock);
+
+            const stage = self.sync_stage.updating_superblock;
+
+            self.state_machine_opened = false;
+            self.state_machine.reset();
+
+            // Bump commit_max before the superblock update so that a view_durable_update()
+            // during the sync_start update uses the correct (new) commit_max.
+            self.commit_max = std.math.max(stage.target.checkpoint_op, self.commit_max);
+
+            self.sync_message_timeout.stop();
```

Comment:
Any expectation as to timeout state before stopping?

Reply:
Good catch! Yes, the timer should always be running.